packages/agents/src/tests/index.test.ts

```diff
@@ -0,0 +1,672 @@
+      const agentName = "my-agent-with-special-chars-123!@#";
+      const agent = await getAgentByName(env.TEST_AGENT, agentName);
+
+      expect(agent).toBeDefined();
+      expect(agent).toHaveProperty("fetch");
+    });
+
+    it("should handle options parameter", async () => {
+      const agentName = "my-agent";
+      const options = { locationHint: "wnam" as DurableObjectLocationHint };
+
+      const agent = await getAgentByName(env.TEST_AGENT, agentName, options);
+
+      expect(agent).toBeDefined();
+      expect(agent).toHaveProperty("fetch");
+    });
+
+    it("should create different agents for different names", async () => {
+      const agent1 = await getAgentByName(env.TEST_AGENT, "agent1");
+      const agent2 = await getAgentByName(env.TEST_AGENT, "agent2");
+
+      expect(agent1).toBeDefined();
+      expect(agent2).toBeDefined();
+      // They should be different instances based on ID
+      const id1 = env.TEST_AGENT.idFromName("agent1");
+      const id2 = env.TEST_AGENT.idFromName("agent2");
+      expect(id1.toString()).not.toBe(id2.toString());
+    });
+  });
+
+  describe("routeAgentRequest", () => {
+    it("should handle CORS preflight requests when cors is enabled", async () => {
+      const request = new Request("http://localhost/agents/TestAgent/test", {
+        method: "OPTIONS",
+        headers: {
+          Origin: "http://localhost:3000",
+          "Access-Control-Request-Method": "POST",
+          "Access-Control-Request-Headers": "Content-Type"
+        }
+      });
+
+      const response = await routeAgentRequest(request, env, { cors: true });
+
+      expect(response).toBeDefined();
+      expect(response!.status).toBe(200);
+      expect(response!.headers.get("Access-Control-Allow-Origin")).toBe("*");
+      expect(response!.headers.get("Access-Control-Allow-Methods")).toBe(
+        "GET, POST, HEAD, OPTIONS"
+      );
+      expect(response!.headers.get("Access-Control-Allow-Credentials")).toBe(
+        "true"
+      );
+    });
+
+    it("should handle CORS with custom headers", async () => {
+      const customCorsHeaders = {
+        "Access-Control-Allow-Origin": "https://example.com",
+        "Access-Control-Allow-Methods": "GET, POST",
+        "Access-Control-Max-Age": "3600"
+      };
+
+      const request = new Request("http://localhost/agents/TestAgent/test", {
+        method: "OPTIONS",
+        headers: {
+          Origin: "https://example.com"
+        }
+      });
+
+      const response = await routeAgentRequest(request, env, {
+        cors: customCorsHeaders
+      });
+
+      expect(response).toBeDefined();
+      expect(response!.status).toBe(200);
+      expect(response!.headers.get("Access-Control-Allow-Origin")).toBe(
+        "https://example.com"
+      );
+      expect(response!.headers.get("Access-Control-Allow-Methods")).toBe(
+        "GET, POST"
+      );
+      expect(response!.headers.get("Access-Control-Max-Age")).toBe("3600");
+    });
+
```

Comment:
maybe have a test where it fails without cors headers for a diff domain request?